### PR TITLE
[WIP] CLI code

### PR DESCRIPTION
**Pull request by Kmaschta**

### CODE REVIEW

File: cli/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "request": "^2.79.0",
+    "traverse": "^0.6.6",
+    "uuid": "^3.0.1"
+  },
+  "devDependencies": {
+    "eslint": "^3.16.0",
+    "eslint-config-airbnb": "^14.1.0",
+    "eslint-plugin-import": "^2.2.0",
+    "eslint-plugin-jsx-a11y": "^4.0.0",
+    "eslint-plugin-react": "^6.10.0"
```
Comment (floo51) [inline cli/package.json]: react ?

File: cli/src/commands/add.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    ${dim('Examples')}
+        comfy add development -f /config/comfy.json
+        comfy add production -c api -t stable -f /config/api.json
+`);
+    ui.exit(code);
+};
+
+
+module.exports = (ui, modules) => function* (args) {
+    const env = args[0];
```
Comment (floo51) [inline cli/src/commands/add.js]: Use array destructuring ?

File: cli/src/commands/add.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const stats = fs.statSync(filename);
+    if (!stats.isFile()) {
+        ui.error(`The object located at ${red(options.f)} is not a file.`);
+        ui.exit(1);
+    }
+
+    const file = fs.readFileSync(filename, 'utf-8');
+
+    let parsedContent;
+    const guessedYAMLFormat = ['.yml', '.yaml'].includes(path.extname(filename));
```
Comment (floo51) [inline cli/src/commands/add.js]: Seems like we try to find a kind of YAML format, rename is to `isYAML` ?

File: cli/src/commands/get.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+const minimist = require('minimist');
+const { parseFLAT, toJSON, toYAML, toEnvVars } = require('../format');
```
Comment (floo51) [inline cli/src/commands/get.js]: `FLAT` is not an acronym, use `Flat` ?

File: cli/src/commands/get.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        --hash      Get a specific configuration version (if the hash is specified, the tag is ignored)
+        --json      Display the configuration as a JSON file
+        --yml       Display the configuration as a YAML file
+        --envvars   Display the configuration as a sourceable bash file
+`);
+    ui.exit(0);
+};
+
+module.exports = (ui, modules) => function* (args) {
+    const env = args[0];
```
Comment (floo51) [inline cli/src/commands/get.js]: Array destructuring ?

File: cli/src/commands/ls.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    ${dim('Options')}
+        help        Show this very help message
+        -a          Show all configuration versions (even non-tagged ones)
+`);
+    ui.exit(code);
+};
+
+module.exports = (ui, modules) => function* (args) {
+    const env = args[0];
```
Comment (floo51) [inline cli/src/commands/ls.js]: Array destructuring ?

File: cli/src/domain/config.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        } catch (error) {
+            ui.printRequestError(error);
+            ui.exit(1);
+        }
+    };
+
+    const add = function* (project, env, content, { configName, tag }) {
+        const body = toFLAT(content);
+
+        // Encrypt all values
```
Comment (floo51) [inline cli/src/domain/config.js]: Unnecessary comment ?

File: cli/src/domain/project.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        const filename = `${process.cwd()}${path.sep}${CONFIG_PATH}`;
+
+        if (!fs.existsSync(filename)) {
+            checkProjectInfos(envs);
+            return envs;
+        }
+        const file = fs.readFileSync(filename, 'utf-8');
+        const config = ini.parse(file);
+
+        const projectInfos = Object.assign({}, envs, {
```
Comment (floo51) [inline cli/src/domain/project.js]: Can you just go like `Object.assign({}, envs, config.project)` ?

File: cli/src/ui/console/index.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
+const chalk = require('chalk');
+const readline = require('readline');
+
+const print = console.log; // eslint-disable-line no-console
+const error = console.error;  // eslint-disable-line no-console
+
+const digestEvent = (argv) => {
+    const args = argv.splice(2); // Remove "node comfy.js"
```
Comment (floo51) [inline cli/src/ui/console/index.js]: Array destructuring ?

File: cli/src/domain/project.js (hunk 8 above)
Comment (Kmaschta) [inline cli/src/domain/project.js]: Nope because of the `id` => `projectId`

File: cli/package.json (hunk 1 above)
Comment (Kmaschta) [inline cli/package.json]: Yep, it's required by eslint-config-airbnb

Comment (djhi) [inline cli/package.json]: Use `eslint-config-airbnb-base` ;)

**Pull request merged**